# Hypothesis testing Question 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

Use dataset: 

### Steps to follow:
1. Define H0 and H1 in the question.
2. Make dimensional array.
3. Use chi square test to determine if there is a relationship between femle male buyer ratios. 
4. Define the result based on the p-value.

### Hypothesis Definition

H0 : Independence of catagorical variables. i.e defective % does not vary by centre.

H1 : Not all proportions are equal. i.e defective % varies by the centre.

alpha : 0.05

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as ss

In [2]:
Q4= pd.read_csv('C:\\Users\\Vignesh R Babu\\Documents\\ExcelRPython\\Assignment Codes\\3\\customer_order_form.csv')
Q4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

### Getting value counts for each country

In [22]:
errorfree = []
defective = []

for i in range(0,4):
    a,b = Q4.iloc[:,i].value_counts() #Returns counts for errorfree and defective 
    errorfree.append(a) 
    defective.append(b)
print(errorfree)
print(defective)

[271, 267, 269, 280]
[29, 33, 31, 20]


### Making a dimensional Array

In [23]:
arr = np.array([errorfree,defective]) #Dimensional Array
arr

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

### Chi square test:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html

In [24]:
ss.chi2_contingency(arr) # OP is in the format : chi2,p-value,dof,expected

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

### Result
p-value = 0.2771 > 0.05. Therefore, we accept the Null Hypotheis that defective % does not vary by centre.i.e. Independence of categorical variables.